Plots/Meteograms/NCL_meteo_1.py

```diff
@@ -0,0 +1,291 @@
+
+# Add a pad between the y axis label and the axis spine
+ax1.yaxis.labelpad = 4
+
+# Rotate the labels on the x axis so they are vertical
+for tick in ax1.get_xticklabels():
+    tick.set_rotation(90)
+
+# Make an axis to overlay on top of the contour plot
+axin = fig.add_subplot(311)
+
+# Use the geocat.viz function to set axes limits and ticks
+gvutil.set_axes_limits_and_ticks(axin,
+                                 xlim=[taus[0], taus[-1]],
+                                 ylim=[levels[0], levels[-1]],
+                                 xticks=np.array(taus),
+                                 yticks=np.linspace(1000, 400, 8),
+                                 xticklabels=np.array(taus),
+                                 yticklabels=np.array(levels))
+
+# Make axis invisible
+axin.patch.set_alpha(0.0)
+axin.axis('off')
+
+# Set aspect ratio of axin so it lines up with axis underneath (ax1)
+axin.set_aspect(0.052)
+
+# Plot wind barbs
+barbs = axin.barbs(taus,
+                   np.linspace(1000, 400, 8),
+                   ugrid,
+                   vgrid,
+                   color='k',
+                   lw=0.2,
+                   length=5)
+
+# Create text box at lower right of contour plot
+ax1.text(1.0,
+         -0.35,
+         "CONTOUR FROM -20 TO 60 BY 10",
+         horizontalalignment='right',
+         transform=ax1.transAxes,
+         bbox=dict(boxstyle='square, pad=0.15',
+                   facecolor='white',
+                   edgecolor='black'))
+
+# Create two inset axes in the second axes
+axin1 = ax2.inset_axes([0.0, 0.3, 1, 0.30])
+axin2 = ax2.inset_axes([0.0, -.1, 1, 0.35])
+
+# Plot bar chart
+
+# Set aspect ratio of inset axis
+axin1.set_aspect(30)
+
+# Plot bars depicting the rain03 variable
+axin1.bar(taus,
+          rain03,
+          width=3,
+          color='limegreen',
+          edgecolor='k',
+          linewidth=.2)
+
+# Use the geocat.viz function to set the y axis label
+gvutil.set_titles_and_labels(axin1,
+                             ylabel='3hr rain total',
+                             labelfontsize=12)
+
+# Determine the labels for each tick on the x and y axes
+yticklabels = ['0.0', '0.10', '0.20', '0.30', '0.40', '0.50']
+xticklabels = ['12z', '', '18z', '', 'Apr29', '',
+               '06z', '', '12z', '', '18z', '',
+               'Apr30', '', '06z', '', '12z', '',
+               '18z', '', 'May01', '', '06z', '', '12z']
+
+# Use the geocat.viz function to set axes limits and ticks
+gvutil.set_axes_limits_and_ticks(axin1,
+                                 xlim=[0, 72],
+                                 ylim=[0, .5],
+                                 xticks=np.arange(0, 75, 3),
+                                 yticks=np.arange(0, .6, 0.1),
+                                 xticklabels=xticklabels,
+                                 yticklabels=yticklabels)
+
+# Use the geocat.viz function to add minor ticks
+gvutil.add_major_minor_ticks(axin1,
```

Comment:
The rain total plot doesn't have minor xticks. You can remove them by specifying `x_minor_per_major=1`